### ISSUE TITLE
We need an attribute that designates that one type can map from another

### ISSUE THREAD

**Issue by undiwahn** [SkywardApps/popcorn#33]
We get a lot of mapping boilerplate lines when using simple maps that don't require much customization.

eg

```
config.Map<Car,CarProjection>();
config.Map<Employee,EmployeeProjection>();
...
```

Ideally, we want an attribute that you can apply that designates something as the target of a mapping.  So for example:

```
[ExpandFrom(typeof(Car), "[My,Default,Includes]"]
class CarProjections {
...
}
```

This means that we need a new method on PopcornConfiguration called something like 'ScanAssemblyForMapping' that will look for all class types within an assembly that have the [ExpandFrom] attribute applied, and automatically register them.

**Comment by alexbarbato** [comment]
Did we want to be able to declare a set of Default includes with the ExpandFrom attribute? Does seem like this could create conflicts with the default include attribute.

**Comment by undiwahn** [comment]
I think the idea was that it would replace the manual Map<> call entirely.
Since you can optionally provide an include list to the Map<> call, you
should be able to optionally provide an include list to the attribute.

**Comment by undiwahn** [comment]
( and you are correct about the conflict; it's the same as between the
Map<> call and the DefaultInclude attribute. So it should also throw an
exception if you try to provide both )

On Oct 29, 2017 7:01 PM, "Nick Elliott" <nelliott@skywardapps.com> wrote:

I think the idea was that it would replace the manual Map<> call entirely.
Since you can optionally provide an include list to the Map<> call, you
should be able to optionally provide an include list to the attribute.

**Comment by undiwahn** [comment]
Closed with PR #36
